src/scripts/main.js

```diff
@@ -1 +1,52 @@
 'use strict';
+
+const firstPromise = new Promise(function (resolve, reject) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0) {
+      resolve('First promise was resolved');
+    }
+
+    setTimeout(() => {
+      reject(new Error(`First promise was rejected`));
+    }, 3000);
+  });
+});
+
+const secondPromise = new Promise(function (resolve) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 || e.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
+
+const thirdPromise = new Promise(function (resolve) {
+  document.addEventListener('mousedown', (e) => {
+    if (e.button === 0 && e.button === 2) {
```

Comment:
The condition `e.button === 0 && e.button === 2` will never be true because a single mouse event cannot have both button values simultaneously. You might want to implement logic that waits for both left and right clicks separately before resolving this promise.